deploy.py

```diff
@@ -134,11 +143,93 @@ def deploy(ctx, target, mode, version=None):
-    client.update_service(
+    return target_info['name'], task_definition["taskDefinitionArn"]
+
+
+@task(help={
+    "target": "Name of the project you want to deploy on AWS: service or harvester",
+    "mode": "Mode you want to deploy to: development, acceptance or production. Must match APPLICATION_MODE",
+    "version": "Version of the project you want to deploy. Defaults to latest version"
+})
+def deploy(ctx, target, mode, version=None):
+    """
+    Updates the container cluster in development, acceptance or production environment on AWS to run a Docker image
+    """
+
+    # Setup the AWS SDK
+    print(f"Starting AWS session for: {mode}")
+    session = boto3.Session(profile_name=ctx.config.aws.profile_name)
+    ecs_client = session.client('ecs', region_name='eu-central-1')
+
+    target_name, task_definition_arn = register_task_definition(
+        ecs_client,
+        ctx.config.aws.task_role_arn,
+        target,
+        mode,
+        version
+    )
+
+    print("Updating service")
+    ecs_client.update_service(
         cluster=ctx.config.aws.cluster_arn,
-        service=f"{target_info['name']}",
+        service=f"{target_name}",
         taskDefinition=task_definition_arn
     )
+
+
+@task(help={
+    "target": "Name of the project you want migrate on AWS: service or harvester",
+    "mode": "Mode you want to migrate: development, acceptance or production. Must match APPLICATION_MODE",
+    "version": "Version of the project you want to migrate. Defaults to latest version"
+})
+def migrate(ctx, target, mode, version=None):
+    """
+    Executes migration command on container cluser for development, acceptance or production environment on AWS
+    """
+
+    # Setup the AWS SDK
+    print(f"Starting AWS session for: {mode}")
+    session = boto3.Session(profile_name=ctx.config.aws.profile_name)
+    ecs_client = session.client('ecs', region_name='eu-central-1')
+
+    target_name, task_definition_arn = register_task_definition(
+        ecs_client,
+        ctx.config.aws.superuser_task_role_arn,
+        target,
+        mode,
+        version
+    )
+
+    print("Migrating")
+    ecs_client.run_task(
+        cluster=ctx.config.aws.cluster_arn,
+        taskDefinition=task_definition_arn,
+        launchType="FARGATE",
+        overrides={
+            "containerOverrides": [{
+                "name": "search-portal-container",
+                "command": ["python", "manage.py", "migrate"],
+                "environment": [
+                    {
+                        "name": "POL_DJANGO_POSTGRES_USER",
+                        "value": f"{ctx.config.django.postgres_user}"
+                    },
+                    {
+                        "name": "POL_SECRETS_POSTGRES_PASSWORD",
+                        "value": f"{ctx.config.aws.postgres_password_arn}"
+                    },
+                ]
+            }]
+        },
+        networkConfiguration={
```

Comment:
Hm, die id's zouden nog wel eens kunnen wijzigen als terraform een destroy -> create doet. Ik weet zo snel alleen ook niet een betere oplossing. Komt later wel een keer (of niet 😉 ).

Reply:
Names is een alternatief? Er zit alleen nog een hoop "dev" in de naamgeving.